pinot-core/src/main/java/org/apache/pinot/core/data/table/TableResizer.java

```diff
@@ -347,7 +368,16 @@ public List<IntermediateRecord> trimInSegmentResults(GroupKeyGenerator groupKeyG
       }
     }
 
-    return Arrays.asList(heap);
+    for (int i = heap.length; i > 0; i--) {
```

Comment:
(MAJOR) This can cause overhead for other scenarios where sort is not required. Let's add a boolean to control whether to sort the result